source/write.txt

```diff
@@ -13,3 +27,137 @@ Write Data to MongoDB
+
+Insert One 
+----------
+
+The following code shows how to insert a single document into a collection:
+
+.. literalinclude:: /includes/usage-examples/write-code-examples.c
+   :start-after: start-insert-one
+   :end-before: end-insert-one
+   :language: c
+   :copyable:
+   :dedent:
+
+To learn more about the ``mongoc_collection_insert_one()`` function, see the
+:ref:`Insert Documents <c-write-insert>` guide.
+
+Insert Multiple
+---------------
+
+The following code shows how to insert multiple documents into a collection:
+
+.. literalinclude:: /includes/usage-examples/write-code-examples.c
+   :start-after: start-insert-many
+   :end-before: end-insert-many
+   :language: c
+   :copyable:
+   :dedent:
+
+To learn more about the ``mongoc_collection_insert_one()`` function, see the
+:ref:`Insert Documents <c-write-insert>` guide.
+
+Update One
+----------
+
+The following code shows how to update a single document in a collection by
+creating or editing a field:
+
+.. literalinclude:: /includes/usage-examples/write-code-examples.c
+   :start-after: start-update-one
+   :end-before: end-update-one
+   :language: c
+   :copyable:
+   :dedent:
+
+To learn more about the ``mongoc_collection_insert_one()`` function, see the
+:ref:`Update Documents <c-write-update>` guide.
+
+Update Multiple
+---------------
+
+The following code shows how to update multiple documents in a collection by
+creating or editing a field:
+
+.. literalinclude:: /includes/usage-examples/write-code-examples.c
+   :start-after: start-update-many
+   :end-before: end-update-many
+   :language: c
+   :copyable:
+   :dedent:
+
+To learn more about the ``mongoc_collection_insert_many()`` function, see the
+:ref:`Update Documents <c-write-update>` guide.
+
+Delete One
+----------
+
+The following code shows how to delete a single document in a collection:
+
+.. literalinclude:: /includes/usage-examples/write-code-examples.c
+   :start-after: start-delete-one
+   :end-before: end-delete-one
+   :language: c
+   :copyable:
+   :dedent:
+
+To learn more about the ``mongoc_collection_delete_one()`` function, see the
+:ref:`Delete Documents <c-write-delete>` guide.
+
+Delete Multiple
+---------------
+
+The following code shows how to delete multiple documents in a collection:
+
+.. literalinclude:: /includes/usage-examples/write-code-examples.c
+   :start-after: start-delete-many
+   :end-before: end-delete-many
+   :language: c
+   :copyable:
+   :dedent:
+
+To learn more about the ``mongoc_collection_delete_many()`` function, see the
+:ref:`Delete Documents <c-write-delete>` guide.
+
+Bulk Write 
+----------
+
+The following code shows how to perform multiple write operations in a single
+bulk operation:
+
+.. literalinclude:: /includes/usage-examples/write-code-examples.c
+   :start-after: start-bulk-write
+   :end-before: end-bulk-write
+   :language: c
+   :copyable:
+   :dedent:
+
+To learn more about the ``mongoc_collection_bulk_operation_with_opts()``
```

Comment:
S: if you edit the code example to use mongoc_bulk_operation_execute(), I'd mention that function here instead
```suggestion
To learn more about the ``mongoc_bulk_operation_execute()``
```